### ISSUE TITLE
Type safety lost on assignment or spread to Pick<> type

### ISSUE THREAD

**Issue by jdforsythe** [Microsoft/TypeScript#29287]
**TypeScript Version:**  3.1.6

**Search Terms:**
pick, spread, known properties

**Code**

```ts
interface IdEntity { id: number }
interface Entity extends IdEntity { name: string; }

type ExcludeOnCreate = keyof IdEntity;

type EntityForCreate<T extends IdEntity> = Pick<T, Exclude<keyof T, ExcludeOnCreate>>;

const ent: Entity = { id: 1, name: 'abc' };

// no error
const entForCreateAssign: EntityForCreate<Entity> = ent;
// no error
const entForCreateSpread: EntityForCreate<Entity> = { ...ent };

// "object literal may only specify known properties..."
const entForCreateError: EntityForCreate<Entity> = { id: 1, name: 'def' };

// Property 'id' does not exist on type Pick<Entity, "name">
console.log(entForCreateAssign.id);

// Property 'id' does not exist on type Pick<Entity, "name">
console.log(entForCreateSpread.id);

//

const testFn = <T extends IdEntity>(ent: EntityForCreate<T>): void => {
    console.log(ent);
};

// argument of type { id: number } is not assignable to parameter of type Pick<Entity, "name">
testFn<Entity>({ id: 1 });

// no error
testFn<Entity>(ent);
```

**Expected behavior:**
The assignment of a variable with extra properties should throw an error. The spread of a variable with extra properties should throw an error. Passing an invalid object as a parameter should throw an error.

**Actual behavior:**
Error is only thrown when assigning object literal or passing object literal to function.

**Playground Link:**
http://www.typescriptlang.org/play/#src=interface%20IdEntity%20%7B%20id%3A%20number%20%7D%0D%0Ainterface%20Entity%20extends%20IdEntity%20%7B%20name%3A%20string%3B%20%7D%0D%0A%0D%0Atype%20ExcludeOnCreate%20%3D%20keyof%20IdEntity%3B%0D%0A%0D%0Atype%20EntityForCreate%3CT%20extends%20IdEntity%3E%20%3D%20Pick%3CT%2C%20Exclude%3Ckeyof%20T%2C%20ExcludeOnCreate%3E%3E%3B%0D%0A%0D%0Aconst%20ent%3A%20Entity%20%3D%20%7B%20id%3A%201%2C%20name%3A%20'abc'%20%7D%3B%0D%0A%0D%0A%2F%2F%20no%20error%0D%0Aconst%20entForCreateAssign%3A%20EntityForCreate%3CEntity%3E%20%3D%20ent%3B%0D%0A%2F%2F%20no%20error%0D%0Aconst%20entForCreateSpread%3A%20EntityForCreate%3CEntity%3E%20%3D%20%7B%20...ent%20%7D%3B%0D%0A%0D%0A%2F%2F%20%22object%20literal%20may%20only%20specify%20known%20properties...%22%0D%0Aconst%20entForCreateError%3A%20EntityForCreate%3CEntity%3E%20%3D%20%7B%20id%3A%201%2C%20name%3A%20'def'%20%7D%3B%0D%0A%0D%0A%2F%2F%20Property%20'id'%20does%20not%20exist%20on%20type%20Pick%3CEntity%2C%20%22name%22%3E%0D%0Aconsole.log(entForCreateAssign.id)%3B%0D%0A%0D%0A%2F%2F%20Property%20'id'%20does%20not%20exist%20on%20type%20Pick%3CEntity%2C%20%22name%22%3E%0D%0Aconsole.log(entForCreateSpread.id)%3B%0D%0A%0D%0A%2F%2F%0D%0A%0D%0Aconst%20testFn%20%3D%20%3CT%20extends%20IdEntity%3E(ent%3A%20EntityForCreate%3CT%3E)%3A%20void%20%3D%3E%20%7B%0D%0A%20%20%20%20console.log(ent)%3B%0D%0A%7D%3B%0D%0A%0D%0A%2F%2F%20argument%20of%20type%20%7B%20id%3A%20number%20%7D%20is%20not%20assignable%20to%20parameter%20of%20type%20Pick%3CEntity%2C%20%22name%22%3E%0D%0AtestFn%3CEntity%3E(%7B%20id%3A%201%20%7D)%3B%0D%0A%0D%0A%2F%2F%20no%20error%0D%0AtestFn%3CEntity%3E(ent)%3B

**Related Issues:**

**Comment by weswigham** [comment]
```
// no error
const entForCreateAssign: EntityForCreate<Entity> = ent;
// no error
const entForCreateSpread: EntityForCreate<Entity> = { ...ent };
```
that's to be expected - a type would always be assignable to a type less one field. If you want to forbid assignments like this, you need to make that field verboten by including it in the type, but of a different type, for example:
```ts
type EntityForCreate<T extends IdEntity> = Pick<T, Exclude<keyof T, ExcludeOnCreate>> & {[K in ExcludeOnCreate]?: undefined};
```
which excludes `id` like you're already doing, but then also adds back an optional `id` field, which, if it exists, must be `undefined`.
